### ISSUE TITLE
[Bug] Discord freezing when in-game overlay is enabled

### ISSUE THREAD

**Issue by PriscillAAH** [BetterDiscord/BetterDiscord#1463]
### Before Reporting

- [X] I found no existing issues matching my bug
- [X] My issue is not caused by a plugin or theme

### Describe the Bug

Whenever I open any game with in-game overlay enabled my discord freezes. Unable to do anything in Discord, it's just completely frozen for about 30 seconds or more, until the overlay appears. I have done clean install Discord and Better Discord. Doesn't freeze on regular Discord  or BD without in-game overlay installed. Still freezes without plug-ins on BD as well.

### Expected Behavior

Expect in-game overlay to open and Discord not to freeze. 

### Steps To Reproduce

1. Install BD
2. Enable in-game overlay
3. Open game 
4. Discord freezes for 30 sec 
5. In-game overlay appears
6. Discord unfreezes 

### Operating System

Windows

### Discord Version

Stable

### BetterDiscord Version

v1.82

**Comment by Tropix126** [comment]
Hmm, BD shouldn't even be touching the overlay at all since it doesn't inject into it. Have you made sure that this only happens with BD installed?

**Comment by PriscillAAH** [comment]
Yes, BD fresh install no plugins. It only freezes when the in-game overlay is turned on for a game with BD installed. Normal Discord + in-game overlay doesn't freeze. https://discord.com/channels/86004744966914048/1028636567792795668 Other people mentioning the same issue as me.

**Comment by Wryyyong** [comment]
I can attest to this same issue, for what it's worth

**Comment by rauenzi** [comment]
I am unable to recreate this. I can attest that BD does not touch the overlay browser at all.

**Comment by NotxFound** [comment]
@rauenzi 

https://user-images.githubusercontent.com/83174921/209886658-16104fa5-6659-4d46-98a2-9e536d5274b2.mp4

An example video, when the game starts discord is working fine, but only when in game the overlay starts to load in, discord freezes till in game the overlay information popup, when the overlay loading is done discord is working again.

I hope I will bring something

**Comment by NotxFound** [comment]
Any updates?

**Comment by rauenzi** [comment]
There is no update as I do not experience this issue nor does anyone in the developer group. BD does not touch anything with the overlay so I have no idea why this would happen.

**Comment by PriscillAAH** [comment]
Issue still happening for me. Could be in too many discords be the issue?

**Comment by skittles-fivem** [comment]
Yeah same thing happens to me especially playing war thunder trying to tab out to watch his stream

**Comment by rauenzi** [comment]
@PriscillAAH is this still happening for you? I tested this again yesterday and was unable to reproduce it.

**Comment by PriscillAAH** [comment]
Yeah, it still is happening for me any time I open games with the overlay enabled it freezes like this. Are there some sort of logs I can give you?

**Comment by rauenzi** [comment]
Not really, BetterDiscord does no touch or inject into the overlay so there's not really any logs to collect. I guess in the desktop client you could record the devtools console to see if there's any obvious spot where it freezes.

**Comment by rauenzi** [comment]
To clarify these steps:
> 
    Install BD
    Enable in-game overlay
    Open game
    Discord freezes for 30 sec
    In-game overlay appears
    Discord unfreezes

Can you say _which_ Discord freezes? As in main window versus the overlay. When you open a game, is the main Discord client freezing for 30 seconds? If so which games does this occur with? I have been actively using the overlay with no effect on the main client window.

**Comment by Wryyyong** [comment]
> Can you say _which_ Discord freezes? As in main window versus the overlay.

The main Discord client window.

> When you open a game, is the main Discord client freezing for 30 seconds?

Roughly 20-30 seconds, yes.

> If so which games does this occur with?

Game-agnostic; any application for which the overlay is enabled will cause this issue when the overlay tries to inject itself, only ceasing once it finishes injection.

**Comment by rauenzi** [comment]
I tested this again just now and neither the overlay nor main window froze for me. I was able to use and interact with the main window the entire time without any lag. I'm not sure what could be causing it for those that experience it, but Discord's overlay is a blackbox that we neither touch nor have control of.

**Comment by Huderon** [comment]
I'm a dummy, but I am able to replicate this issue fairly consistently as long as I have hardware acceleration enabled. It seems to occur specifically while the overlay is connecting to the game. I have no idea how to troubleshoot beyond that point though. 

Interestingly, I experience the issue all the same on native Discord. Although it doesn't seem quite as intense, there's still a noticeable pause while the overlay connects with hardware acceleration is enabled

**Comment by rauenzi** [comment]
I also have hardware acceleration enabled but I don't notice this with or without BD. Any change you could post your PC's specs for comparison?

**Comment by Huderon** [comment]
It's odd, for sure. As it stands, I can definitely still notice it without BD installed, even if it's not quite as bad.

For those that are experiencing it, I'd suggest disabling hardware acceleration in the 'Advanced' section of your Discord settings to see if that changes anything for you
